runtime/caml/domain.h

```diff
@@ -129,9 +129,8 @@ int caml_domain_is_in_stw();
 void caml_run_on_all_running_domains_during_stw(void (*handler)(struct domain*, void*), void* data);
 int caml_try_run_on_all_domains_with_spin_work(void (*handler)(struct domain*, void*, int, struct domain**), void*, void (*leader_setup)(struct domain*),
   void (*enter_spin_callback)(struct domain*, void*), void*,
-  void (*leave_spin_callback)(struct domain*, void*), void*,
-  int);
-int caml_try_run_on_all_domains(void (*handler)(struct domain*, void*, int, struct domain**), void*, void (*leader_setup)(struct domain*), int);
+  void (*leave_spin_callback)(struct domain*, void*), void*);
```

Comment:
Given that there is no longer a leave spin in the API, we should also remove `leave_spin_callback` as that is unused.